.travis.yml

```diff
@@ -44,7 +54,16 @@ jobs:
       script:
         - build/makerelease.sh
         # wildcard copy to allow us to move versioning forward without updating travis
+        # NOTE: we run these here to verify a release is ready. the deploy step will run
+        # again, instead of us copying the artifacts around S3
         - cp -v libarchive-*.tar.gz .versity/SOURCES/
         - (cd .versity; docker build -f Dockerfile.build -t versity/libarchive-build .)
         - (cd .versity; bash ./build_rpm_in_docker.sh)
         - (cd .versity; docker build -f Dockerfile -t versity/python:2.7-libarchive .)
+    - stage: deploy
```

Comment:
This stage seems to rebuild artifacts instead of re-using them; is that the intent?

Reply:
correct -- first validating we can build them, then a final build and upload. Sharing artifacts between stages is A Thing I didn't care to tackle yet.

Reply:
Cool, I agree with that model of it (building is a type of validation!); not sure of Travis' long-term artifact strategy either.